RELEASE/scripts/excavator/projects/x_genie.ash

```diff
@@ -0,0 +1,94 @@
+/**
+ * @author gausie
+ * Determine which monsters and effects can be fought/acquired with the Genie.
+ */
+
+import <excavator/x_utils.ash>
+
+void spade_genie( string page )
+{
+    // Check page validity
+    if ( !page.contains_text( "<b>Results:</b>" ) )
+    {
+        return;
+    }
+
+    // Find our wish or bail out
+    matcher m = "You announce, \"I wish (.*?)\"<".create_matcher( page );
+
+    if ( !m.find() )
+    {
+        return;
+    }
+
+    string wish = m.group( 1 );
+
+    // Determine if this was a success
+    boolean success = !page.contains_text( "<br />You could try, " );
+
+    string [string] data = {
+        "success": success,
+    };
+
+    boolean wishable;
+
+    matcher fight_matcher = "(?:to (?:fight|be fighting) (?:a )?)(.*)".create_matcher( wish );
+
+    // Determine wishability for fight and prepare data map
+	if ( fight_matcher.find() )
+    {
+        monster mon = fight_matcher.group( 1 ).to_monster();
+
+        data["type"] = "monster";
+        data["value"] = mon.to_string();
+
+        wishable = !mon.boss && mon.copyable;
+    }
+    // Determine wishability for effect and prepare data map
+    else if ( wish.starts_with( "to be " ) || wish.starts_with( "i was " ) )
+	{
+		effect eff = wish.substring( 6 ).to_effect();
+
+        data["type"] = "effect";
+        data["value"] = eff.to_string();
```

Comment:
We may want to add whether mafia thinks it's wishable, here? That way we don't have to cross-reference old versions of mafia on the backend to determine whether the person is just using an old version with an incorrect database entry?

Reply:
when i start the data map above, it contains whether this attempt was successful or not, and thus we can infer whether their version of mafia thought it should or not